### PR TITLE
Shared_ptr<UTransport> nullptr check

### PR DESCRIPTION
**Pull request by ruchirchauhan**
All code that takes a shared_ptr<UTransport> parameter, now check for nullptr.

### CODE REVIEW

File: src/communication/NotificationSource.cpp [hunk 1]
```diff
@@ -35,11 +39,19 @@ NotificationSource::NotificationSource(
 v1::UStatus NotificationSource::notify(
     datamodel::builder::Payload&& payload) const {
 	auto message = notify_builder_.build(std::move(payload));
+	if (!transport_) {
```
Comment (gregmedd) [inline src/communication/NotificationSource.cpp:42]: Is this check necessary here? Is it possible to have a null transport_ member if it's checked before construction?

**Comment by gregmedd** [comment]
I'm wondering if this shouldn't target the 1.0 branch so it can be included in the 1.0.1 bugfix release

**Comment by gregmedd** [comment]
@ruchirchauhan - please rebase this PR to the `v1.0_up-v1.6.0` branch. I have already updated the merge target for the PR

File: src/communication/NotificationSource.cpp (hunk 1 above)
Comment (ruchirchauhan) [reply src/communication/NotificationSource.cpp]: It's not I agree. Removed

**Review by gregmedd — APPROVED** [review]

**Pull request merged**